### ISSUE TITLE
Error on First Run (ERROR root: Valid login required, exiting)

### ISSUE THREAD

**Issue by mikebanks** [hangoutsbot/hangoutsbot#898]
## Description
Attempted to do first run and DEFINITELY putting in the correct creds and receive this error.

## Steps to reproduce:
  1. Followed installation instructions
  2. first run command: "python3 hangupsbot/hangupsbot.py"

## Log output

```
user@box: python3 hangupsbot/hangupsbot.py
01:50:59 INFO config: /home/ubuntu/.local/share/hangupsbot/config.json read
01:50:59 INFO root: memory = /home/ubuntu/.local/share/hangupsbot/memory.json, failsafe = 3, delay = 1
01:50:59 INFO config: /home/ubuntu/.local/share/hangupsbot/memory.json read
01:50:59 INFO requests.packages.urllib3.connectionpool: Starting new HTTPS connection (1): accounts.google.com
Sign in with your Google account:
Email: email@gmail.com	
Password: 
01:51:12 ERROR root: LOGIN FAILED
Traceback (most recent call last):
  File "/home/ubuntu/.local/lib/python3.5/site-packages/hangups/auth.py", line 132, in get_auth
    raise GoogleAuthError("Refresh token not found")
hangups.auth.GoogleAuthError: Refresh token not found

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/ubuntu/.local/lib/python3.5/site-packages/hangups/auth.py", line 231, in _get_authorization_code
    return browser.get_cookie('oauth_code')
  File "/home/ubuntu/.local/lib/python3.5/site-packages/hangups/auth.py", line 199, in get_cookie
    return self._session.cookies[name]
  File "/home/ubuntu/.local/lib/python3.5/site-packages/requests/cookies.py", line 282, in __getitem__
    return self._find_no_duplicates(name)
  File "/home/ubuntu/.local/lib/python3.5/site-packages/requests/cookies.py", line 340, in _find_no_duplicates
    raise KeyError('name=%r, domain=%r, path=%r' % (name, domain, path))
KeyError: "name='oauth_code', domain=None, path=None"

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "hangupsbot/hangupsbot.py", line 137, in login
    cookies = hangups.auth.get_auth_stdin(cookies_path)
  File "/home/ubuntu/.local/lib/python3.5/site-packages/hangups/auth.py", line 64, in get_auth_stdin
    return get_auth(CredentialsPrompt(), refresh_token_cache)
  File "/home/ubuntu/.local/lib/python3.5/site-packages/hangups/auth.py", line 138, in get_auth
    session, credentials_prompt
  File "/home/ubuntu/.local/lib/python3.5/site-packages/hangups/auth.py", line 233, in _get_authorization_code
    raise GoogleAuthError('Authorization code cookie not found')
hangups.auth.GoogleAuthError: Authorization code cookie not found
01:51:12 ERROR root: Valid login required, exiting
```

**Comment by EmuMan44** [comment]
Unless you have the email 'email@gmail.com' available to you (which I seriously doubt), then you need to input the login info you gave to the bot in the email and password section.  This should NOT be your current account, I would recommend creating a new one, something like 'emubot69@gmail.com' (that's what I did), logging on with it, and connecting it to the hangouts page.  Then, do the exact same thing but type in the correct email and password.  Just in case you didn't know, you then have to type "/bot iamspartacus" in the hangouts chat and it will initialize you as an admin.  You should be good then!

**Comment by mikebanks** [comment]
So @EmuMan44 yes of course I don't have that email address. I didn't include it in here for reasons... :-) I included the correct gmail and the correct creds and it produced that error.

**Comment by oboote** [comment]
tl;dr solution:
1. Login manually: https://github.com/tdryer/hangups/issues/350#issuecomment-323553771
2. Copy ~/.cache/hangups/refresh_token.txt to ~/.local/share/hangupsbot/cookies.json

Struggled getting bot to login for a while before eventually finding workaround.

Suspected issue:
Unrecognised/untrusted new login is being served an additional challenge which the bot is unable to complete (captcha, etc.)

Attempted workarounds which didn't help:
- SSH tunneling to server and login using a browser from server IP.
- Login from server using lynx
- Opening https://accounts.google.com/displayunlockcaptcha locally and whilst tunneling.

**Comment by Achran** [comment]
In case you're using terminal only server like me, here's workaround which worked for me:
1. Download w3m
2. open google.com in w3m
3. login with your bot credentials
4. go through the additional security check (I prefer to set recovery email and then verify by it)
5. Close the w3m and rerun the bot

**Comment by Terrance** [comment]
hangups now provides a manual login flow option, which can be used to simplify the workaround steps (reading cookies in the browser) without downloading a standalone script.
